File: familycare-front-end/src/app/principal/consulta/consulta.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from "@angular/core";
import { Observable } from 'rxjs';
import { API_URL } from './../../ambiente/ambiente';
import { Consulta } from './consulta';

@Injectable()
export class ConsultaService {

  constructor(private http: HttpClient){}

    salvar(consulta: Consulta){
        return this.http.post(`${API_URL}consultas`, consulta);
    }

    listarTodos(){
        return this.http.get(`${API_URL}consultas`);
    }

    listarComPaginacao(pagina:number,tamanho:number){
        return this.http.get(`${API_URL}consultas/${pagina}/${tamanho}`);
    }

    listarComFiltro(pagina:number,tamanho:number,descricaoFilter:string){
        return this.http.get(`${API_URL}consultas/${pagina}/${tamanho}/${descricaoFilter}`);
    }

    delete(id:string){
        return this.http.delete(`${API_URL}consultas/${id}`);
    }

    consultarPorId(id:number){
        return this.http.get(`${API_URL}consultas/${id}`);
    }

}

File: familycare-back-end/src/main/java/br/ufg/familycare/service/ConsultaService.java
package br.ufg.familycare.service;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import br.ufg.familycare.model.Consulta;
import br.ufg.familycare.repository.ConsultaRepository;

@Service
public class ConsultaService {

	@Autowired
	private ConsultaRepository consultaRepository;

	public Consulta salvar(Consulta consulta) {
		return consultaRepository.save(consulta);
	}

	public Optional<Consulta> consultarPorId(Long id) {
		return consultaRepository.findById(id);
	}

	public Iterable<Consulta> listarTodos(Long userId) {
		return consultaRepository.findByUsuarioIdOrderById(userId);
	}

	public Page<Consulta> listarComPaginacao(int pagina, int tamanho, Long userId) {
		Pageable pageable = PageRequest.of(pagina, tamanho, Sort.Direction.ASC, "id");
		return this.consultaRepository.findByUsuarioIdOrderById(pageable, userId);
	}

	public Page<Consulta> listarComFiltro(int pagina, int tamanho, String descricaoFilter, Long userId) {
		Pageable pageable = PageRequest.of(pagina, tamanho, Sort.Direction.ASC, "id");
		return this.consultaRepository.findByDescricaoIgnoreCaseContainingAndUsuarioIdOrderById(pageable,
				descricaoFilter, userId);
	}

	public void deletarPorId(Long id) {
		consultaRepository.deleteById(id);
	}

}